src/rustzcash.rs

```diff
@@ -1,7 +1,97 @@
 extern crate libc;
 extern crate sapling_crypto;
+extern crate pairing;
 
-use libc::uint64_t;
+#[macro_use]
+extern crate lazy_static;
+
+use pairing::{
+    BitIterator,
+    PrimeFieldRepr,
+    PrimeField,
+    bls12_381::{
+        Bls12,
+        Fr,
+        FrRepr
+    }
+};
+
+use sapling_crypto::{
+    jubjub::JubjubBls12,
+    pedersen_hash::{
+        pedersen_hash,
+        Personalization
+    }
+};
+
+use libc::{uint64_t, size_t, c_uchar};
+
+lazy_static! {
+    static ref JUBJUB: JubjubBls12 = {
+        JubjubBls12::new()
```

Comment:
Just to confirm, this is only the Jubjub parameters? The Sapling and Sprout CRSs will be loaded/unloaded via pointers? Or will the Sapling CRS be similarly statically-loaded?

Reply:
@ebfull discussed this. The Sapling and Sprout CRSs will be loaded via an API through which zcashd passes the paths to the params. Both verification keys along with the Sapling proving key will be loaded into memory immediately, while the Sprout proving key will be loaded and unloaded on-demand.